File: sparse_lin_reg.m
clearvars
clc;

m = 5e2; n = 5e3; k = 5; sigma = 1e-2;

%% Training set
A_train = randn(m,n); 
x = zeros(n,1); x(randsample(n,k)) = 1;
b_train = A_train*x + sigma*randn(m,1);

%% Test set
A_test = randn(100,n);
b_test = A_test*x;

File: sparse_lin_reg.py
import numpy as np

m = 500; n = 5000; k = 5; sigma = 1e-2

# Training set
A_train = np.random.randn(m,n)
index = np.random.choice(n, k)
x = np.zeros((n,1), dtype=np.float); x[index] = 1.0
b_train = np.matmul(A_train, x) + sigma * np.random.randn(m,1)

# Test set
A_test = np.random.randn(100,n)
b_test = np.matmul(A_test, x)
